packages/services/src/serverconnection.ts

```diff
@@ -245,22 +245,24 @@ namespace Private {
 
     // Use explicit cache buster when `no-store` is set since
     // not all browsers use it properly.
-    let cache = init.cache || settings.init.cache;
+    const cache = init.cache || settings.init.cache;
     if (cache === 'no-store') {
       // https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Using_XMLHttpRequest#Bypassing_the_cache
       url += (/\?/.test(url) ? '&' : '?') + new Date().getTime();
     }
 
-    let request = new settings.Request(url, { ...settings.init, ...init });
+    const request = new settings.Request(url, { ...settings.init, ...init });
 
-    // Handle authentication.
+    // Handle authentication. Authentication can be overdetermined by
+    // settings token and XSRF token.
     let authenticated = false;
     if (settings.token) {
       authenticated = true;
       request.headers.append('Authorization', `token ${settings.token}`);
-    } else if (typeof document !== 'undefined' && document.cookie) {
-      let xsrfToken = getCookie('_xsrf');
-      if (xsrfToken !== void 0) {
+    }
+    if (typeof document !== 'undefined' && document && document.cookie) {
```

Comment:
Can you elaborate on why you took out the `else` here?

Reply:
This was causing reconnects to fail. I didn't definitively understand why but it seems reasonable if we have both the cookie and the header to use both and it turned out to eliminate a class of failures after reconnect.